src/table.ts

```diff
@@ -772,12 +775,18 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
       });
     }
 
+    // If rowKeys and ranges are both empty, the request is a full table scan
```

Comment:
I think we actually want to remove this block of code. I am not sure what it is trying to accomplish, but note that we don't need it since a full table scan happens anyways under these conditions.

Reply:
This section is incorrect: https://github.com/googleapis/nodejs-bigtable/blob/main/src/table.ts#L820-L827

If the original request only had rowKeys and no ranges, it'll add a new range to the original request. It's only correct when the original request is a full table scan.

To simplify the logic here, we add an empty range if the request was a full table scan. So when we're retrying, we don't need to manually add a new range. When it loops through all the ranges, it'll see that end key of empty range: {} is **after** lastRowKey, so it'll add range (startKey, _) to the retry request.